### PR TITLE
CAT-2106 Do not allow email addresses as user names

### PR DESCRIPTION
**Pull request by ndrnour**
easy way to fix this by preventing user from using "@" in the username field
cause Z.B `example@gmail` is understandable AS E-Mail address as well

### CODE REVIEW

**Comment by ndrnour** [comment]
@robertpainsi what's your opinion about what I did here?

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/login/RegistrationDialogFragment.java [hunk 1]
```diff
@@ -125,16 +125,22 @@ private void onRegisterButtonClick() {
...
 			RegistrationTask registrationTask = new RegistrationTask(getActivity(), username, password, email);
 			registrationTask.setOnRegistrationCompleteListener(this);
 			registrationTask.execute();
 		} else {
-			new AlertDialog.Builder(getActivity())
-					.setTitle(R.string.register_error)
-					.setMessage(R.string.register_password_mismatch)
-					.setPositiveButton(R.string.ok, null)
-					.show();
+			showErrorAlertDialog(R.string.register_password_mismatch);
```
Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/login/RegistrationDialogFragment.java]: Please change the order of the if-else statements
```
if ("@" in username)
  show error
else if (not password equals) 
  show error
else
  execute registration task
```

Comment (ndrnour) [reply catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/login/RegistrationDialogFragment.java]: done.

File: catroid/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -902,6 +902,7 @@
     <string name="oauth_username_taken">Someone else already uses this name. Please choose a different one!</string>
     <string name="show_password">Show Password</string>
     <string name="register_password_mismatch">Passwords do not match</string>
+    <string name="register_email_as_username_error">Email address can not used as username</string>
```
Comment (robertpainsi) [inline catroid/src/main/res/values/strings.xml]: Email address can not **be** used as username

**Comment by robertpainsi** [comment]
The web team also just checks for the @ sign https://github.com/Catrobat/Catroweb-Symfony/pull/123/files#diff-7f7ef968389cbeb3ea2934cb53172debR21

File: catroid/src/main/res/values/strings.xml (hunk 2 above)
Comment (ndrnour) [reply catroid/src/main/res/values/strings.xml]: done

**Comment by ndrnour** [comment]
I've fixed this in other PR [2819](https://github.com/Catrobat/Catroid/pull/2819)

**Pull request closed without merging**